### ISSUE TITLE
"Error opening the library"

### ISSUE THREAD

**Issue by nickv2002** [YACReader/yacreader#35]
I'm trying t set up a new library running on a small VNC session that uses XFCE running on Ubuntu Server 18.04. I installed the Ubuntu repo and pulled the appropriate package to install, so far so good. Creating a library mostly works (small error is that typing into the name box thinks my keyboard layout is something completely different but I was able to copy/paste in) and the scan proceeds showing my library of comics being scanned. However once that's done (or if I stop the scan) I get the following screen:
![yacreader error](https://user-images.githubusercontent.com/704392/50733088-77133180-113b-11e9-92ad-256aea494579.jpg)
That error message appears to say "Error opening the library". I get the same message if I close YACReader Library and re-open. What's going on here? Any log file I can check?

**Comment by luisangelsm** [comment]
Hi,

How did you install YACReader? Could it be that some packages are missing?
Check if you have the Qt's sql little plugin installed.

Best regards

**Comment by nickv2002** [comment]
I installed by adding the repository and grabbing the package like this:
```
sudo sh -c "echo 'deb http://download.opensuse.org/repositories/home:/selmf/xUbuntu_18.04/ /' > /etc/apt/sources.list.d/home:selmf.list"
wget -nv https://download.opensuse.org/repositories/home:selmf/xUbuntu_18.04/Release.key -O Release.key
sudo apt-key add - < Release.key
sudo apt-get update
sudo apt-get install yacreader
```
Here' the libqtsql packages I have installed:
```
apt list --installed|grep libqt.sql
WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
libqt5sql5/bionic,now 5.9.5+dfsg-0ubuntu1 amd64 [installed,automatic]
libqt5sql5-sqlite/bionic,now 5.9.5+dfsg-0ubuntu1 amd64 [installed,automatic]

```
Are there other libqt5sql packages that should be installed? I noticed it complained about not having libqr to on the screen where it shows the QR code for example.

**Comment by nickv2002** [comment]
@luisangelsm looks like I have the right QT MYSQL files installed (see previous comment ⬆️) Anywhere I could find a log file with more info on what's going wrong?

**Comment by selmf** [comment]
My guess would be that this is either a corrupt library or a path problem, possibly related to the "small problem" with the locale that you described.

Log and config files are located at ~/.local/share/YACReader/YACReaderLibrary/ but will probably not be very helpful as the log level is hardcoded to error only.

**Comment by nickv2002** [comment]
I tried it again using new VMs installed from scratch on Xubuntu 18.04 and Mint Cinnamon 19.1
On both the 'small problem' of weird text entry was not an issue anymore so 👍 to that at least.

However I hit the same error once I created a library. Here's Mint's error message:
![mint](https://user-images.githubusercontent.com/704392/51093929-34fe7700-175d-11e9-8f61-6c054ef13c9d.jpg)
Checking the log on revealed this one error line:
`ERROR 2019-01-13T17:57:10.639 Unable to create data base "Driver not loaded-Driver not loaded"`
🙁 Any ideas on why YACReaderLibrary won't work any of these disros?

**Comment by selmf** [comment]
Not really. The database driver should be shipped with the libqt5sql5-sqlite package and we do have reports of users successfully using the Deb packages.

If you provide me some info on the Vms you used I can run some tests to see what the actual problem is.

**Comment by nickv2002** [comment]
For my attempt to set up a YACReader server VM, here's what I did:

1. Install a base version of _Ubuntu Server 18.04.1 LTS (64 Bit)_ then run all the apt updates so everything was up-to-date.
1. Set my Mylar-sorted comics library to [mount via CIFS/SAMBA with `cifs-utils`](
https://askubuntu.com/questions/157128/proper-fstab-entry-to-mount-a-samba-share-on-boot/157140#157140).
1. Since Ubuntu Server doesn't come with any GUI, I followed this [detailed guide from Digital Ocean](https://www.digitalocean.com/community/tutorials/how-to-install-and-configure-vnc-on-ubuntu-18-04#step-1-—-installing-the-desktop-environment-and-vnc-server) to adding an XFCE environment and VNC support.
2. Finally, I installed YACReader using the Debian Package repository like so:

```
sudo sh -c "echo 'deb http://download.opensuse.org/repositories/home:/selmf/xUbuntu_18.04/ /' > /etc/apt/sources.list.d/home:selmf.list"
wget -nv https://download.opensuse.org/repositories/home:selmf/xUbuntu_18.04/Release.key -O Release.key
sudo apt-key add - < Release.key
sudo apt-get update
sudo apt-get install yacreader
```
Thanks for your help @selmf

**Comment by selmf** [comment]
>     2. Set my Mylar-sorted comics library to [mount via CIFS/SAMBA with `cifs-utils`](https://askubuntu.com/questions/157128/proper-fstab-entry-to-mount-a-samba-share-on-boot/157140#157140).

And there we have the culprit. Using YACReader on a network file system is not really recommended. The problem is that sqlite relies on a mechanism called file locking to prevent database corruption. This mechanism is buggy in most network file systems, which can lead to interesting bugs.

So far the only solution that I am aware of is to completely disable file locking via the "nolock" mount option. Word of caution: This will disable file locking for **all** files, so make sure you know what you are doing or you might find yourself with a few problems ;)

**Comment by nickv2002** [comment]
Is there really no way for me to store the DB on the local (virtual) disk while pointing at the media on an external CIFS volume?

Also why make GUI for the server app rather than just using a web UI or CLI?

I guess these were historical decisions that made sense at the time or for different use cases. But it seems YACReader is just the wrong software for my use case (I want to download comics on a headless *nix server, and read on iOS).

Thanks for your help though @selmf

**Comment by selmf** [comment]
Well, I wouldn't say that there is no way at all to do this. YACReader stores everything in a hidden .yacreaderlibrary folder in the root directory of your library. You could just create a directory on your local disk and symlink it to the network share like this:

> ln -s /path/to/local/database /path/to/library/.yacreaderlibrary

We also have a CLI version, called the headless server, but that is not part of the packages so you'd have to compile it yourself.

**Comment by luisangelsm** [comment]
Before any other steps, please, create a small library locally so you can
exlude from the equation the mounted shared folder. Check it
YACReaderLibrary is able to create the library.

Where do you have your comics stored? YACReaderLibraryServer (command line)
can run in arm devices, and there are users that are running it in their
NAS systems.

YACReaderLibrary existed before the server, so including the server in it
was a natural thing to do.

I use YACReaderLibraryServer in a Raspberry Pi with a hard disk atrached...

**Comment by nickv2002** [comment]
Thanks for your suggestions. I will give the CLI version a try (when I have some tinkering time free soon) as that's exactly what I'm looking for. I didn't know it existed because it's not mentioned on the [download page](http://www.yacreader.com/downloads).

**Comment by luisangelsm** [comment]
In ios, use the help section to contact me.

Thanks for the feedback!

**Comment by luisangelsm** [comment]
About the port, the app tries to use the default port, but if it is in use
it picks a random port, after the firts collition it should use the same
port in the next start....

You can set the port in the YACReaderLibrary.ini file, I do not remember
the full path to that file, you can find it in the users' forum.

**Comment by Frederick888** [comment]
Sorry to bump up an old thread. Just want to say that instead of `mfsymlinks`, I can use YAC with `mount.cifs -o user,uid=1000,gid=1000,rw,nounix,nobrl` and no linking is required.
